### PR TITLE
add a diff engine

### PR DESCRIPTION
**Pull request by tobowers**
This adds a simple diff-er to the hamt. I checked in with the IPLD folks, and they couldn't think of a more efficient way to do this... which means we probably need a different data structure long term, but for now it should suit our needs.

The need FYI - is in community for the nodes to support each other if one misses a add block request here and there.

### CODE REVIEW

File: differ/diff.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/quorumcontrol/go-hamt-ipld"
+	"github.com/quorumcontrol/go-hamt-ipld/goipldpb"
+	"github.com/quorumcontrol/go-hamt-ipld/pb"
+	"golang.org/x/xerrors"
+)
+
+// FindNew takes an existing HAMT and returns the new Key/Value pairs found in the newHamt
+func FindNew(ctx context.Context, cs *hamt.CborIpldStore, existingHamt *hamt.Node, newHamt *hamt.Node) ([]*pb.KV, error) {
+	respChan := make(chan *pb.KV)
```
Comment (zonotope) [inline differ/diff.go]: this looks like an orphaned channel. i see where it's closed below, but it isn't returned from the enclosing function and nothing within the function is putting anything onto the channel or reading from it.

Comment (cap10morgan) [inline differ/diff.go]: What does `respChan` do in here? Looks like it just gets closed when the first nodes are equal, but nothing else references it.

**Review by cap10morgan — COMMENTED** [review]
Question about `respChan` in `differ/diff.go`. The rest looks good to me.

**Review by zonotope — APPROVED** [review]
looks like there's a dangling channel, but :+1: otherwise.

File: differ/diff.go (hunk 1 above)
Comment (tobowers) [reply differ/diff.go]: oops - thanks for seeing this. I was originally going to stream differences through a channel, but that gets too complicated in a recursive way, so removed it (most of it apparently).

**Pull request merged**